### DISCUSSION TITLE
KOG Story OUTDATED

### DISCUSSION THREAD

**Discussion by slysnivy** [shoc71/dont-stop-now#319] [General]
# The story:
- Our alien guy is making a delivery shipment of star squares (they’re like animals or pets)
- He makes a pit stop by a fuel station to refuel 
- For some reason, the star spirits are agitated as we hear loud banging and clanging. Our player goes into the rocket ship to check
- Right as he sees the cage, it bursts open and all of the star squares flood out of the ship and into the wild

# Our Player and Level Menus
Our player has two modes, speed mode (with maybe some jet shoes) used to play through the levels and normal mode for navigating through the hub. We could say that speed mode is quick but sacrifices controlled horizontal movement (Only W controls). Meanwhile normal mode allows for normal walking (controlled horizontal movement, includes WASD controls)

- In terms of levels and menus, there will be a hub zone for each level SET. The hub zone will introduce an NPC (with a unique attitude) and will always hint of the locations of the star squares (only for the sake of story) and could choose to give out a quest (under specific conditions, for instance the NPC could ask, “you’ll need to catch 20 star squares in this area to prove that you’ll be able to fetch this package for me”)
- The hub area will act as an area for choosing levels in that level set via moving to the left/right side of the screen, AND a way to move in between level sets using a Map (map on a big sign board)
- The player will unlock the next level set if they reached the end of the area 

# Playing through the levels and Star Square (Starmet) behaviour
- In terms of the star squares being in the level, they have 3 behaviours
1. They will only move within the predefined space
2. They will actively run away if you get too close
3. Otherwise, if no player is near, they will randomly move from time to time
- the player can not choose or actively decide to collect all of the star squares, as it won’t be required to move onto the next level set and hub

# Possible Endings (depending on how much you unlock)
- There are 3 outcomes when the player finishes the game:
1. they can leave the planet without collecting all of the star squares, to which it kind of gives a generic/lame ending where you made the shipment on time, but you got scolded for not collecting all of the star squares (what this does: game will encourage you to collect all of the star squares)
2. you collect all of the star squares and you get praised on delivering all of the star squares on time. Everyone’s happy, but you wonder what caused the cage break in the first place, because star squares are usually docile and peaceful (what this does: game will make you look for something, maybe by then re visiting each NPC in their hub zones across level sets. This will unlock a new set of dialogue where NPC’s give more hints into the history of star squares and their importance to their culture/planet. we can have them discuss this additional dialogue earlier in the game, like when they completed their quest)
3. You collect all of the star squares, in addition to finding “level 0” on your FIRST play through! it’s kind of a gate to an ancient star square temple but has been sealed away.

- Usually on a first playthrough, most players are expected to go through 1 -> 2 and then -> 3, but there will be a chance that some players will experience to 2 or 3 (depending on how much progress they make on the first playthrough) without experiencing the previous ones
- Outcome 3 (where all playthroughs end up) will bring the game to its natural conclusion. 

# The main ending
- We find out after unlocking level 0 that a cultist used the star squares for “star energy” to summon a boss that was sealed. . Turns out the star squares are used as a catalyst to open/close the prison that the boss was sealed in. After this scene, you will run deeper into the ruins/temple to tire out the boss (no more star energy) and defeat it once in for all.

# Link to visual storyboard:
https://docs.google.com/presentation/d/10o0QlCjiDWyNN6-QNkA6Ij3n4rRahgT5DrM3E4ab-xg/edit?usp=sharing

**Comment by shoc71** [comment]
- Bad guy steals and captures our stars / we must chase him because those are OUR stars (responsibility about this like his role can be altered later)
- We get hints about this guy's location through the NPCs that are helping us and a lot of the stars escaping on their own, so we must capture them as well
- We finally catch-up to the guy but he doesn't have any of the stars, he claims to have hid them all '**somewhere**'
- We go to that location only to find out the other stars has been used as keys to release the **Big Bad Boss** - the **Big Bad Boss** has been summoned
- The **Big Bad Boss** will try and consume us because we possess a lot of those stars
- Going through the temple levels in reverse (making mirror copy of levels)
- To beat the boss, the player either pass all the levels or survive until the time runs out (time is determined by the total number of stars - number of stars collected)
- Purpose of levelzero - is for different endings and potentially gather other stars into your possession and add them (120% completion of the game)

**Comment by shoc71** [comment]
Big Bad Boss Behavior
- ### charge attack
- (faster than player, ignores any block collision, one direction only, stops when he collides with screen or hits player)
- ### free roam
- (chases the player unconditionally, ignores any block collision, slower than player speed)

**Reply by slysnivy** [reply]
The boss will switch between these two behaviors

**Reply by shoc71** [reply]
yes, thanks for mentioning that. Both velocity and timing to switch between behaviors will be tested at a later date. I think to add urgency to some more of the other levels rather than keeping him in a loop, is to have a timer counting down at the top/bottom where it just starts raining lava or something like that.
